### DISCUSSION TITLE
Feature request - Auto Updating

### DISCUSSION THREAD

**Discussion by DeFlanko** [xnorpx/blue-onyx#5] [Ideas]
Really hoping someone can make a Docker for this.

**Comment by xnorpx** [comment]
Hi @DeFlanko welcome to Blue Onyx

Can you explain the docker request and why it would be useful?

**Reply by DeFlanko** [reply]
Getting updates via the Docker Desktop is mainly the reason; long term maintenance -- however I'm open for any new process this application will handle updates. 

Also, while your here -- how to i hook back into Blue Iris? I'm trying to find documentation on it but cant seem to find any.

**Reply by xnorpx** [reply]
Updates on Windows will probably best case be an installer that runs a service. 

I don't think Docker on Windows for the effort of replacing 3 files, but it is a good point to provide an easy way to allow for updates.

You need to start the service: blue_onyx.exe --port <some port>

Then in blue iris you point to 127.0.0.1:<port> under ai configuration.

**Reply by xnorpx** [reply]
I should do some guides regarding this in the discussion. Easy to be blind once you know how to do it.

**Comment by DeFlanko** [comment]
I went back to your other repo and found the lenghtly discussion on what folks have done already... i think im caught up... 

Here is what ive done -- alternative to installing as a service. Seems to be working thus far. 

## Create a "Service Account" 

In my case i'll use my same service account as Blue iris: **BI_SVC** 


## Create a BAT file

Firstly, I saved Blue Onyx in its own folder on my secondary drive `E:\` -- Adjust your bat file accordingly

Create a BAT file and name it `RunBlueOnyx.bat`

Edit in Notepad: 
```batch
start "" "E:\Blue Onyx\blue_onyx.exe" --port 32168 --gpu-index 0 --log-level info --log-path "E:\Blue Onyx"
```
the default `--port 32168` and `--gpu-index 0` are being enforced; easier for troubleshooting later


## Create a Task Scheduler Job

Name: **Auto Start Blue Onyx AI**

Triggers: **At System Startup**

Actions:  **Start a program - <Path to BAT file>** (in my case `E:\Blue Onyx\RunBlueOnyx.bat`)

Run the task as: `<YOUR_SERVICE_ACCOUNT>` in my case `BI_SVC`

## Validate in Task Manager

![image](https://github.com/user-attachments/assets/4bbe86c6-841b-49a7-9cb8-9e2e9c02d7e4)

**Comment by DeFlanko** [comment]
one thing i wasn't sure on is how to make the log file... do i have the syntax correct? is it just the path to save to? or is it file name too?

**Reply by xnorpx** [reply]
it should be path only

**Comment by xnorpx** [comment]
@DeFlanko 

I have created a powershell script to install/update blue onyx

I also implemented the is /v1/status/updateavailable so it should report back that there is new version available if there is a new version available on github

**Comment by DeFlanko** [comment]
If i make a Pull request it's not going to auto accept the changes right?

**Reply by xnorpx** [reply]
No, needs sign-off and passing ci for merge. So feel free to send pr.
